test/integration/model/upsert.test.js

```diff
@@ -570,6 +570,20 @@ describe(Support.getTestDialectTeaser('Model'), () => {
         });
       }
 
+      if (dialect === 'mysql' || dialect === 'mariadb') {
+        it('should allow to use calculated values on duplicate', async function () {
+          const insertUser = await this.User.create({
+            bar: 10,
+          });
+          const [user] = await this.User.upsert({
+            id: insertUser.id,
+            bar: this.sequelize.literal('`bar` + 1'),
```

Comment:
It would be interesting to test whether this works when there was no duplicate (if there isn't already a test for that).